serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java

```diff
@@ -289,6 +292,29 @@ private static boolean checkOutsideMaxAge(
     return timeDifference > maxAge.getSeconds();
   }
 
+  /**
+   * Populate histogram metrics that can be used for analysing online retrieval calls
+   *
+   * @param entityRows entity rows provided in request
+   * @param featureReferences feature references provided in request
+   * @param project project name provided in request
+   */
+  private void populateHistogramMetrics(
+      List<GetOnlineFeaturesRequestV2.EntityRow> entityRows,
+      List<FeatureReferenceV2> featureReferences,
+      String project) {
+    Metrics.requestEntityCount.labels(project).observe(Double.valueOf(entityRows.size()));
+    Metrics.requestFeatureCount.labels(project).observe(Double.valueOf(featureReferences.size()));
+
+    Set<String> featureTableRefs =
+        featureReferences.stream()
+            .map(featureReference -> getFeatureTableStringRef(project, featureReference))
+            .collect(Collectors.toSet());
+    Metrics.requestFeatureTableCount
+        .labels(project)
+        .observe(Double.valueOf(featureTableRefs.size()));
```

Comment:
nit: looks like you use featureTableRefs just for counting distinct strings, consider doing
```
long countDistinct = featureReferences.stream().map(...)
                        .distinct()
                        .count();
....
        .observe(Double.valueOf(countDistinct));

```